packages/metro-resolver/src/resolve.js

```diff
@@ -102,6 +103,8 @@ function resolve(
     } while (candidate !== next);
   }
 
+  nodeModulesPaths.push(...context.nodeModulesPaths);
```

Comment:
Let's add helpful comments here explaining why we only push the custom node paths until now

Reply:
good suggestion! what do you think about something like this?
```suggestion
  /**
   * It's important that we push nodeModulesPaths to the list of resolution
   * candidates after the do while loop above.
   *
   * This ensures that, when we resolve a transitive dependencies inside
   * nodeModulesPaths, the hierarchical lookup is still respected and takes
   * precedence over the root of the nodeModulesPaths locations.
   *
   * E.g.
   * When resolving an import in /extra-node-modules/foo/main.js with
   * nodeModulesPaths=["/extra-node-modules"].
   *
   * We want to resolve to modules in /extra-node-modules/foo/node_modules
   * first, before /extra-node-modules.
   */
  nodeModulesPaths.push(...context.nodeModulesPaths);
```

Reply:
FWIW I'd be OK without a super verbose comment here. Something like `// Fall back to `nodeModulesPaths` after hierarchical lookup, similar to $NODE_PATH` would be fine. The commit history will have the rest of the context.

Not totally opposed to the longer comment, but it's the kind of thing that can be hard to keep up to date and correctly anchored as the code evolves.

(OTOH we should absolutely have a page in the docs that goes into detail about the resolution algorithm - I intend to write one soon - and I can see why in the absence of one we'd want to annotate the code more.)

Reply:
Sounds good! I'm always trying to weigh what might confusing to people, but your suggestion works! & the issue commit should help explain more in lieu of that docs page 😅 
```suggestion
  // Fall back to `nodeModulesPaths` after hierarchical lookup, similar to $NODE_PATH
  nodeModulesPaths.push(...context.nodeModulesPaths);
```